Use "phy" or "file" code path depending on parameter type

libvirt always uses "phy" type - it is the only type backed by xen-blkback
module and supported by libxl at the same time.

--- xen-4.4.2/tools/hotplug/Linux/block	2011-03-25 11:42:50.000000000 +0100
+++ xen-4.4.2/tools/hotplug/Linux/block	2011-06-09 13:57:15.000000000 +0200
@@ -232,7 +232,7 @@
             "/local/domain/$FRONTEND_ID/vm" 'unknown')
 
     case $t in 
-      phy)
+      phy|file)
         dev=$(expand_dev $p)
 
         if [ -L "$dev" ]
@@ -240,16 +240,13 @@
           dev=$(readlink -f "$dev") || fatal "$dev link does not exist."
         fi
         test -e "$dev" || fatal "$dev does not exist."
-        test -b "$dev" || fatal "$dev is not a block device."
-
-        claim_lock "block"
-        check_device_sharing "$dev" "$mode"
-	write_dev "$dev"
-        release_lock "block"
-	exit 0
-	;;
-
-      file)
+        if [ -b "$dev" ]; then
+          claim_lock "block"
+          check_device_sharing "$dev" "$mode"
+          write_dev "$dev"
+          release_lock "block"
+          exit 0
+        fi
         # Canonicalise the file, for sharing check comparison, and the mode
         # for ease of use here.
         file=$(readlink -f "$p") || fatal "$p does not exist."
@@ -329,14 +326,12 @@
   remove)
     t=$(cat $HOTPLUG_STORE-type)
     case $t in 
-      phy)
-	exit 0
-	;;
-
-      file)
+      phy|file)
         claim_lock "block"
-        node=$(cat "$HOTPLUG_STORE-node")
-        losetup -d "$node"
+        node=$(cat "$HOTPLUG_STORE-node" 2>/dev/null)
+        if [ -n "$node" ]; then
+          losetup -d "$node"
+        fi
         release_lock "block"
 	exit 0
 	;;
